src/se/vidstige/jadb/managers/InputManager.java

```diff
@@ -0,0 +1,28 @@
+package se.vidstige.jadb.managers;
+
+import se.vidstige.jadb.JadbDevice;
+import se.vidstige.jadb.JadbException;
+import se.vidstige.jadb.Stream;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Author: pablobaxter.
+ */
+public class InputManager {
+
+    private final JadbDevice device;
+
+    public InputManager(JadbDevice device) {
+        this.device = device;
+    }
+
+    public void sendEvent(InputEvent event) throws JadbException {
+        try (InputStream is = device.executeShell("input", event.buildArgs())) {
+            Stream.flushRead(is);
+        } catch (IOException e) {
+            throw new JadbException(e.getMessage());
```

Comment:
Here the important inner exception gets lost, please pass e to `JadbException` constructor.